src/libraries/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -307,6 +307,34 @@ private static bool IsDomainEqualToHost(string domain, string host)
                    (string.Compare(host, 0, domain, 1, host.Length, StringComparison.OrdinalIgnoreCase) == 0);
         }
 
+        internal static int CompareDomain(string? left, string? right)
```

Comment:
Given that both callers of this method only care about equality and not order, this can probably be simplified into `DoaminEquals`.